plugins/catalog/src/components/CatalogTable/CatalogTable.tsx

```diff
@@ -96,6 +97,16 @@ export const CatalogTable = (props: CatalogTableProps) => {
     }
   }, [filters.kind?.value]);
 
+  useEffect(() => {
+    const tableColumns = columns ?? defaultColumns;
+    const entityFields = tableColumns
+      .map(column => column.field)
+      .filter(field => field?.startsWith('entity.'))
+      .map(field => field!.substring('entity.'.length)) as string[];
+    // relations are used in the resolved fields below
+    addEntityFields(['relations', ...entityFields]);
```

Comment:
This won't remove fields we no longer care about in case of an update right? Only add new ones.

I wonder if we should wrap things up a little bit more so that we're able to handle that?

Ideas 😅 :

```tsx
const { createFieldRegistration } = useEntityList()
const fieldRegistration = useMemo(() => createFieldRegistration(), [createFieldRegistration])

useEffect(() => () => fieldRegistration.unregister(), [])
useEffect(() => fieldRegistration.setFields(extraFields), [fieldRegistration, extraFields.join('\0')])
```

But probably a bit more neatly wrapped into `useEntityList` somehow.

The `setFields` method would replace the existing fields, and the full set of fields is the set of fields from all current registrations.

Can probably get rid of the `unregister` method and bake that into a `setFields` equivalent, just wanted to highlight that.